projects/jsf-validation/src/lib/schema-validation.service.ts

```diff
@@ -0,0 +1,88 @@
+import Ajv, { ErrorObject } from 'ajv';
+import { isNil } from 'lodash-es';
+import { JSFJsonSchema } from '../../../jsf/src/lib/jsf-json-schema';
+
+export class SchemaValidationService {
+  private static properties = 'properties';
+  private static missingProperty = 'missingProperty';
+  private static nameProperty = 'name';
+  private static isHidden = 'isHidden';
+  private static isReadOnly = 'isReadOnly';
+
+  /**
+   * Validate the JSON data against the schema using AJV
+   * @param schema: schema to validate against
+   * @param values: values to validate
+   */
+  public static validate(schema: JSFJsonSchema, values: any): JSFErrorObject[] {
+    const ajv = new Ajv({allErrors: true});
+    const valid = ajv.validate(schema, values);
+    if (!valid) {
+      const errors = ajv.errors.map(error => SchemaValidationService.toJsfError(ajv, error));
+      return errors.filter(error => !error.errorSchema[SchemaValidationService.isHidden])
+        .filter(error => !error.errorSchema[SchemaValidationService.isReadOnly]);

```

Comment:
We might be able to simplify this like this. Just an idea.
```suggestion
     return ajv.errors.map(error => SchemaValidationService.toJsfError(ajv, error))
                               .filter(error => !error.errorSchema[SchemaValidationService.isHidden])
                               .filter(error => !error.errorSchema[SchemaValidationService.isReadOnly]);

```